magicparse/schema.py

```diff
@@ -124,8 +125,13 @@ def key() -> str:
 
 class ColumnarSchema(Schema):
     def get_reader(self, stream: BytesIO) -> Iterable[str]:
-        stream_reader = codecs.getreader(self.encoding)
-        return stream_reader(stream)
+        stream_reader_factory = codecs.getreader(self.encoding)
+        stream_reader = stream_reader_factory(stream)
+        while True:
```

Comment:
Pourquoi cette modification est nécessaire ?

Quand on fait `for row in reader` L74 on itère déjà sur le stream ligne par ligne.
D'ailleurs pourquoi tu précises que le paramètre de readline `keepends=False` ? L'implémentation par défaut la met a True.

Cf: https://github.com/python/cpython/blob/de54cf5be371a6f5e2e9f208c38def5f81d3ef02/Lib/codecs.py#L645

Reply:
Le but de cette modification est de pouvoir passer ce False dans le constructeur pour que le parsing des lignes ne renvoie pas le retour chariot. Actuellement il est à True et on reçoit un "\n" dans row. Dans le cas du CSV il est enlevé et on renvoie un tableau de string mais dans le cas du Columnar il était gardé. Je trouve plus logique que le reader qui parse des lignes enlève le retour charriot.
Vu qu'on utilise dans le parse le fait que le StreamReader soit un Iterable, on appelle soit next soit iterate et c'est trop tard pour passer le false.
Donc le seul moyen que j'ai trouvé pour:
1. Ne pas renvoyer le retour de chariot
2. Conserver un Iterable pour ne pas charger tout le fichier en mémoire
c'est de faire cette boucle avec un readline.

Reply:
Ou est le test unitaire qui représente le cas que tu souhaites gérer ?

Reply:
Ouai je serais curieux aussi de voir l'effet via un TU du fait de ne pas renvoyer le retour chariot

Reply:
Le test pète sur quel assert, et quel est la différence entre la valeur attendue et la valeur retournée ?

Reply:
Le reader renvoie une liste de string (d'ailleurs les 2 méthodes get_reader des schemas ne renvoient pas le même type de données, c'est ce qui complexifie le problème car sinon j'aurais pu mettre un fixe unique dans parse).
Sans la modif il renvoie:
"8013109C\n"
"\n"  => c'est ça qui pose problème

Avec la modif il renvoie:
"8013109C"
""

Reply:
Ok je comprends !

> d'ailleurs les 2 méthodes get_reader des schemas ne renvoient pas le même type de données

Effectivement elles ne renvoient pas le même type concret, mais les deux type sont des itérables.

Reply:
ok, j'avais pas vu les retour à la ligne dans l'input.